src/services/room-service.spec.js

```diff
@@ -167,17 +173,26 @@ describe('room-service', () => {
     let myPrivateRoom = null;
 
     beforeEach(async () => {
+      userService.getUserById = sandbox.stub().resolves(myUser);
```

Comment:
A better practice is to let the sandbox do the stubbing instead of replacing the function implementation.

`sandbox.stub(userService, 'getUserById').resolves(myUser);`

It's also how we do it in all other tests.

Reply:
For some reason this doesn't work on this file. I've tried both here and on the room lock, any ideas?

Reply:
I've done the investigation locally and found the causes.

**Case 1**
`sandbox.stub(userService, 'getUserById').resolves(myUser);` not working

This is because `setupTestUser` is called in `beforeAll`, but `pruneTestEnvironment` is called in `afterEach`.

What you need to do is:
- move both `setupTestUser` calls in a `beforeEach`
- remove `userService` and it's mocking altogether, as it is not needed (that's the whole point of `setupTestUser`)

**Case 2:**
`sandbox.stub(roomLockStore, 'takeLock').resolves(lock);` not working

The cause of this is that you did not add an equivalent `afterEach` for that particular test set, restoring the sandbox.

Reply:
Fixed, thanks.